paimon-core/src/main/java/org/apache/paimon/table/FileStoreTableFactory.java

```diff
@@ -83,6 +83,10 @@ public static FileStoreTable create(
         }
         if (writeMode == WriteMode.APPEND_ONLY) {
             table = new AppendOnlyFileStoreTable(fileIO, tablePath, tableSchema);
+        } else if (writeMode == WriteMode.TABLE) {
+            // BatchFileStoreTable need set write_only option to disable compaction
+            dynamicOptions.set(CoreOptions.WRITE_ONLY.key(), "true");
```

Comment:
I prefer disable compaction for `AppendOnlyFileStoreWrite`. Or just create a `BatchFileStoreWrite`.